### PR TITLE
SpriteBodyComponent

### PR DESCRIPTION
**Pull request by spydon**
# Description

Introduces a SpriteBodyComponent that will keep track of how your sprite should be aligned to the underlying body.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/box2d/sprite_body_component.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  @override
+  bool loaded() => body.isActive() && spriteComponent.loaded();
+
+  @override
+  void render(Canvas c) {
+    super.render(c);
+    if (spriteComponent.loaded()) {
+      final screenPosition = viewport.getWorldToScreen(body.position);
```
Comment (spydon) [inline lib/box2d/sprite_body_component.dart:33]: If this one is done in update() the rendering of the sprite lags behind the rendering of the fixtures more and more the more speed the body has. Any clue why this is?

File: doc/box2d.md [hunk 2]
```diff
@@ -8,14 +8,18 @@ There is a simple example game [here](https://github.com/feroult/haunt) that can
...
 If you are going to use Box2D in your project it can be a good idea to use the Box2D specific extension of the `BaseGame` class.
 
-It is called `Box2DGame` and it will control the adding and removal of Box2D's BodyComponents from your root `Box2DComponent`.
+It is called `Box2DGame` and it will control the adding and removal of Box2D's BodyComponents as well as your normal components.
 
 A simple `Box2DGame` implementation example can be seen in the [examples folder](/doc/examples/box2d/simple).
 
+## SpriteBodyComponent
+
+See [SpriteBodyComponent](/doc/components.md#) under Components.
```
Comment (luanpotter) [inline doc/box2d.md:21]: feel free to move it here if you think it's more organized (and leave just a link back on that file).

File: doc/examples/box2d/sprite_body/.gitignore [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Miscellaneous
```
Comment (luanpotter) [inline doc/examples/box2d/sprite_body/.gitignore:1]: do we still need this file? I thought @renancaraujo made a generic one for all examples

File: lib/box2d/sprite_body_component.dart (hunk 1 above)
Comment (luanpotter) [reply lib/box2d/sprite_body_component.dart:33]: I assume it's an order of operations problem, if make this on the update when does it happen in relation to the box2d update? should probably be box2d update -> your viewport conversion -> render

**Comment by erickzanardo** [comment]
Is there anything missing for this to be merged?

**Pull request merged**